### PR TITLE
Test CompoundCommand

### PR DESCRIPTION
**Pull request by CharlotteXiaoYeDing**

### CODE REVIEW

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        aCompoundCommand3 = new CompoundCommand();
+        aCompoundCommand3.add(aCompoundCommand1);
+        aCompoundCommand3.add(aCompoundCommand2);
+        
+        aCompoundCommand4 = new CompoundCommand();
+        aCompoundCommand4.add(aCompoundCommand1);
+        aCompoundCommand4.add(aMoveCommand2);
+        aCompoundCommand4.add(aDeletNodeCommand);
+    }
+
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: Please don't use underscores in method names. You can just do the camelCase version.

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        assertTrue(aGraph.getEdges().contains(aCallEdge));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsSimpleCommands() 
+    {
+        aCompoundCommand1.execute();
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: you can make the test simple by removing lines 105-117. There's no reason the NodesToBeRemoved list should be true at this point.

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        catch (IllegalArgumentException | IllegalAccessException e1) 
+        {
+            fail();
+        }
+    }
+    
+    @Test
+    public void testExecute_ContainsCompoundCommands() 
+    {
+    	assertFalse(aGraph.getRootNodes().contains(aClassNode));
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: Same comment, the `assertFalse` predicates can be removed, and assumed. Given the `setup` method, you can start the test directly at line 143 (the command execution).

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+        assertEquals(aActorNode.getBounds(), new Rectangle2D.Double(0, 0, aActorNode.getBounds().getWidth(), aActorNode.getBounds().getHeight()));
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsCompoundCommands() 
+    {
+        aCompoundCommand3.execute();
+        
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(2, 8, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: same comment, you can remove the baseline predicates (everything until the SUT execution, i.e., the call to `undo`

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    private Field aNodesToBeRemoved;
+    private Field aEdgesToBeRemoved;
+    private Field aNeedsLayout;
+    private Field aRootNodes;
+    private AddNodeCommand aAddClassNodeCommand;
+    private AddNodeCommand aAddActorNodeCommand;
+    private AddEdgeCommand aAddEdgeCommand;
+    private MoveCommand aMoveCommand1;
+    private MoveCommand aMoveCommand2;
+    private DeleteNodeCommand aDeletNodeCommand;
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: use refactoring to fix the typo `Delet` -> `Delete`

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        {
+            fail();
+        }
+    }
+    
+    @SuppressWarnings("unchecked")
+	@Test
+    public void testExecute_ContainsBothCommands() 
+    {
+    	assertFalse(aGraph.getRootNodes().contains(aClassNode));
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: Same comment: start the test with the SUT execution, line 208

File: test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+        }
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsBothCommands() 
+    {
+        aCompoundCommand4.execute();
+        
+        try
```
Comment (prmr) [inline test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java]: remove the baseline predicates, until the call to `undo`

**Comment by CharlotteXiaoYeDing** [comment]
changes made

**Pull request merged**